sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotCallGCSuppressFinalize.NetCore.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Tests.Diagnostics
+{
+    public class Custom : IAsyncDisposable
+    {
+        public async ValueTask DisposeAsync()
```

Comment:
Can you add two more cases with True Positives that will invoke `GC.SuppressFinalize(this)`?
* Method `Something` in this class
* Another class that does NOT implement `IAsyncDisposable` but has `public void DisposeAsync()` with the same name